lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -254,17 +265,10 @@ func (ic *ICache) RemoveImageByConfig(imageConfig *metadata.ImageConfig) {
 	}
 
 	prefixedID := prefixImageID(imageConfig.ImageID)
-	if _, ok := ic.cacheByID[prefixedID]; ok {
-		delete(ic.cacheByID, prefixedID)
-	} else {
-		log.Debugf("Not found in cache by id: %s", prefixedID)
-	}
+	delete(ic.cacheByID, prefixedID)
+	delete(ic.cacheByName, imageConfig.Reference)
 
-	if _, ok := ic.cacheByName[imageConfig.Reference]; ok {
-		delete(ic.cacheByName, imageConfig.Reference)
-	} else {
-		log.Debugf("Not found in cache by name: %s", imageConfig.Reference)
-	}
+	ic.dirty = true
```

Comment:
Is it better to have save called in delete and add method? So user does not have to know if cache should be persisted.